### PR TITLE
 feat(connector): [Adyen] implement Adyen bank transfers and voucher payments in Indonesia  

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
A bank transfer is the most popular payment method in the Indonesian market, second to traditional cash transactions. For every order we create a virtual account (VA) so that the shopper can make a bank transfer into that account.
As an alternative to a bank transfer, shoppers can complete their ecommerce transaction by paying cash at a convenience store. Because Indonesia is still a largely cash-based society, this is a popular payment method.

Bank Transfers Payment Method Types
```
permata_bank_transfer
bca_bank_transfer
bni_va
bri_va
cimb_va
danamon_va
mandiri_va
```
Voucher Payment Method Types
```
alfamart
indomaret
```
Payment Method Data
Bank Transfer
```
"payment_method": "bank_transfer",
  "payment_method_type": "{{Bank_Transfers_Payment_Method_Type}}",
  "payment_method_data": {
    "bank_transfer": {
      "{{Bank_Transfers_Payment_Method_Type}}": {
        "billing_details": {
          "first_name": "Some",
          "second_name": "one",
          "email": "someone@something.com"
        }
      }
    }
  }
```
Voucher
```
"payment_method": "voucher",
  "payment_method_type": "{Voucher_Payment_Method _Types}}",
  "payment_method_data": {
    "voucher": {
      "{{Voucher_Payment_Method}}": {
        "first_name": "Some",
        "second_name": "one",
        "email": "someone@something.com"
      }
    }
  }
```
Output

Bank Transfer
```
"next_action": {
            "type": "display_bank_transfer_information",
            "bank_transfer_steps_and_charges_details": {
                "doku_bank_transfer_instructions": {
                    "expires_at": "2023-07-29T00:10:00",
                    "reference": "8856079400139050",
                    "instructions_url": "https://www.doku.com/how-to-pay/bank-permata.php"
                },
                "receiver": null
            }
   }
```

Voucher
```
  next_action": {
            "type": "display_voucher_information",
            "voucher_details": {
                "expires_at": "2023-07-29T00:10:00",
                "reference": "8888823200139049",
                "download_url": null
            }
      }
```

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Screen Shot 2023-07-27 at 10 40 34 PM

Screen Shot 2023-07-27 at 10 40 49 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -1198,11 +1265,13 @@ pub struct BoletoVoucherData {
 #[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, serde::Deserialize, ToSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum VoucherData {
-    Boleto(Box<BoletoVoucherData>),
```
Comment (ArjunKarthik) [inline crates/api_models/src/payments.rs:1201]: Please keep box for enum values

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -1396,14 +1465,16 @@ pub struct BankTransferNextStepsData {
...
 
 #[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize, ToSchema)]
 pub struct VoucherNextStepData {
-    /// Reference number required for the transaction
+    // Voucher expiry date and time
+    pub expires_at: Option<String>,
+    // Reference number required for the transaction
     pub reference: String,
-    /// Url to download the payment instruction
+    // Url to download the payment instruction
```
Comment (ArjunKarthik) [inline crates/api_models/src/payments.rs:1477]: use /// for comments

File: crates/router/src/connector/stripe/transformers.rs [hunk 3]
```diff
@@ -2954,6 +2962,10 @@ impl
                     payments::BankTransferData::Pix {} | payments::BankTransferData::Pse {} => Err(
                         errors::ConnectorError::NotImplemented("payment method".to_string()).into(),
                     ),
+                    _ => Err(errors::ConnectorError::NotImplemented(
```
Comment (ArjunKarthik) [inline crates/router/src/connector/stripe/transformers.rs:2965]: Do not use default, Let have all the applicable arms

File: crates/router/src/connector/adyen/transformers.rs [hunk 4]
```diff
@@ -237,10 +237,9 @@ pub struct AdyenThreeDS {
 #[derive(Debug, Clone, Deserialize)]
 #[serde(untagged)]
 pub enum AdyenPaymentResponse {
-    Response(Response),
-    PresentToShopper(AdyenPtsResponse),
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:241]: Let's keep separate struct as per the use case and refactor all your changes in this file

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**